src/Datadog.Trace.ClrProfiler.Managed/Integrations/Testing/NUnitIntegration.cs

```diff
@@ -342,20 +183,148 @@ private static void FinishScope(Scope scope, object testExecutionContext, Except
             }
 
             execute(workShift);
-            SynchronizationContext context = SynchronizationContext.Current;
+            AutoInstrumentation.Testing.Common.FlushSpans();
+        }
+
+        /// <summary>
+        /// NUnit.Framework.Api.NUnitTestAssemblyRunner.WaitForCompletion() instrumentation
+        /// </summary>
+        /// <param name="testAssemblyRunner">The NUnitTestAssembly instance</param>
+        /// <param name="timeout">The timeout</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <param name="moduleVersionPtr">A pointer to the module version GUID.</param>
+        /// <returns>The original method's return value.</returns>
+        [InterceptMethod(
+            TargetAssembly = NUnitAssembly,
+            TargetType = NUnitTestAssemblyRunnerType,
+            TargetMethod = WaitForCompletionMethod,
+            TargetMinimumVersion = Major3Minor0,
+            TargetMaximumVersion = Major3,
+            TargetSignatureTypes = new[] { ClrNames.Bool, ClrNames.Int32 })]
+        public static object NUnitTestAssemblyRunner_WaitForCompletion(
+            object testAssemblyRunner,
+            int timeout,
+            int opCode,
+            int mdToken,
+            long moduleVersionPtr)
+        {
+            if (testAssemblyRunner == null) { throw new ArgumentNullException(nameof(testAssemblyRunner)); }
+
+            Type testAssemblyRunnerType = testAssemblyRunner.GetType();
+            Func<object, int, object> execute;
+
+            try
+            {
+                execute = MethodBuilder<Func<object, int, object>>
+                    .Start(moduleVersionPtr, mdToken, opCode, WaitForCompletionMethod)
+                    .WithConcreteType(testAssemblyRunnerType)
+                    .WithParameters(timeout)
+                    .WithNamespaceAndNameFilters(ClrNames.Bool, ClrNames.Int32)
+                    .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: NUnitTestAssemblyRunnerType,
+                    methodName: WaitForCompletionMethod,
+                    instanceType: testAssemblyRunnerType.AssemblyQualifiedName);
+                throw;
+            }
+
+            object result = execute(testAssemblyRunner, timeout);
+            AutoInstrumentation.Testing.Common.FlushSpans();
```

Comment:
Should this use the FlushSpans overload that takes an integration?

Reply:
Yes, fixed in 45506f2